servers/src/main/java/tachyon/worker/block/AsyncEvictor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.Pair;
+import tachyon.Sessions;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.AlreadyExistsException;
+import tachyon.exception.InvalidStateException;
+import tachyon.exception.NotFoundException;
+import tachyon.exception.OutOfSpaceException;
+import tachyon.util.CommonUtils;
+
+/**
+ * AsyncEvictor periodically checks if there is enough space reserved on each storage tier, if
+ * there is no enough free space on some tier, free space from it.
+ */
+public class AsyncEvictor implements Runnable {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private final BlockDataManager mBlockManager;
+  /** Mapping from tier alias to space to be reserved on the tier */
+  private final List<Pair<Integer, Long>> mBytesToReserveOnTiers =
+      new ArrayList<Pair<Integer, Long>>();
+  /** Milliseconds between each check */
+  private final int mCheckIntervalMs;
+  /** Flag to indicate if the checking should continue */
+  private volatile boolean mRunning;
+
+  public AsyncEvictor(BlockDataManager blockManager, TachyonConf tachyonConf) {
```

Comment:
In general, we should avoid taking `TachyonConf` as a parameter in constructors. And in your case, it can be retrieved from `WorkerContext.getConf()`.